### PR TITLE
SNOW-1283946 Embed structured types

### PR DESCRIPTION
**Pull request by sfc-gh-pfus**
### Description

SNOW-1283946 Handle embedded structured types.

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: converter.go [hunk 1]
```diff
@@ -1022,6 +1003,153 @@ func arrowToValue(ctx context.Context, rowIdx int, srcColumnMeta execResponseRow
 	return nil, fmt.Errorf("unsupported data type")
 }
 
+func handleNativeArrowMap(ctx context.Context, rowIdx int, keyMetadata, valueMetadata fieldMetadata, srcValue arrow.Array, loc *time.Location, higherPrecision bool, params map[string]*string) (snowflakeValue, error) {
```
Comment (sfc-gh-dheyman-1) [inline converter.go:1006]: i have impression we pass too many values between functions... perhaps we can introduce some kind of conversion context to wrap these that are inseparable?

File: converter.go [hunk 2]
```diff
@@ -851,98 +936,16 @@ func arrowToValue(ctx context.Context, rowIdx int, srcColumnMeta execResponseRow
...
-			})
-		case objectType:
-			return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (*structuredType, error) {
-				m := make(map[string]any, len(srcColumnMeta.Fields[0].Fields))
-				for fieldIdx, field := range srcColumnMeta.Fields[0].Fields {
-					m[field.Name] = values.(*array.Struct).Field(fieldIdx).ValueStr(int(j))
-				}
-				return buildStructuredTypeRecursive(m, srcColumnMeta.Fields[0].Fields, params), nil
-			})
+			return nil, nil
```
Comment (sfc-gh-dheyman-1) [inline converter.go:946]: perhaps we can log that this ocurred?

File: converter.go (hunk 1 above)
Comment (sfc-gh-dheyman-1) [inline converter.go:1006]: `build` instead of `handle`?

File: structured_type_test.go [hunk 3]
```diff
@@ -569,6 +600,114 @@ func TestArrayOfObjects(t *testing.T) {
 	})
 }
 
+func TestArrayOfArrays(t *testing.T) {
+	warsawTz, err := time.LoadLocation("Europe/Warsaw")
+	assertNilF(t, err)
+	testcases := []struct {
+		name     string
+		query    string
+		actual   any
```
Comment (sfc-gh-dheyman-1) [inline structured_type_test.go:609]: this is more of an `actualType`, right?

File: converter.go (hunk 2 above)
Comment (sfc-gh-pfus) [reply converter.go:946]: Why? This is perfectly normal situation, that we have null value from DB.

File: converter.go (hunk 1 above)
Comment (sfc-gh-pfus) [reply converter.go]: I was considering it, but each function has different arguments... I wouldn't introduce it. At the end I'd like to make some refactoring (if I found some time), maybe I will be able to clean it up a bit.

Comment (sfc-gh-pfus) [reply converter.go]: Ok, done. Needed to change the whole name though.

File: structured_type_test.go (hunk 3 above)
Comment (sfc-gh-pfus) [reply structured_type_test.go:609]: Actually no. This is a placeholder for actual value.

File: converter.go (hunk 1 above)
Comment (sfc-gh-dheyman-1) [reply converter.go]: Ok.

Comment (sfc-gh-dheyman-1) [reply converter.go]: Sure, thanks.

**Review by sfc-gh-dheyman-1 — APPROVED** [review]

**Pull request merged**